src/Hidden/Hidden.tsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+import {useMedia} from '../hooks/useMedia'
+import {viewportRanges} from '../hooks/useResponsiveValue'
+
+type Viewports = 'narrow' | 'regular' | 'wide'
+
+type HiddenProps = {
+  on: Array<Viewports> | Viewports
+  children: React.ReactNode
+}
+
+export const Hidden = ({on: hiddenViewports, children}: HiddenProps) => {
+  const isNarrowViewport = useMedia(viewportRanges.narrow)
+  const isRegularViewport = useMedia(viewportRanges.regular)
+  const isWideViewport = useMedia(viewportRanges.wide)
```

Comment:
Do you think it'd be possible to generate one media query string from the given viewports? Would be cool to do something like:

```jsx
function mediaQuery(rangeOrRanges: Array<Viewports> | Viewports) {
  const ranges = Array.isArray(rangeOrRanges) ? rangeOrRanges : [rangeOrRanges]
  return ranges.map(range => viewportRanges[range]).join(', ')
}

export const Hidden = ({on: hiddenViewports, children}: HiddenProps) => {
  const matches = useMedia(mediaQuery(hiddenViewports))
  if (matches) {
    return null
  }
  return children
}
```

Basically any time one of the media queries from `on` is true, we know it should hidden, otherwise return the children.

Reply:
Yeah, its a good optimisation!